### PR TITLE
[SHIRO-750] update jax-rs dependency to jakarta. Non-Breaking change.

### PR DESCRIPTION
**Pull request by bmhm**
Please discuss whether or not this is a breaking change.

From my perspective it is not, feel free to prove me wrong :)

### CODE REVIEW

**Comment by bmhm** [comment]
Travis build log:

https://travis-ci.com/github/bmhm/shiro/builds/155929411

File: support/jaxrs/pom.xml [hunk 1]
```diff
@@ -45,8 +45,8 @@
         </dependency>
 
         <dependency>
-            <groupId>javax.ws.rs</groupId>
-            <artifactId>javax.ws.rs-api</artifactId>
+            <groupId>jakarta.ws.rs</groupId>
+            <artifactId>jakarta.ws.rs-api</artifactId>
```
Comment (bdemers) [inline support/jaxrs/pom.xml]: set the scope to `provided` and we should be good

Comment (bmhm) [reply support/jaxrs/pom.xml]: Done.

**Comment by fpapon** [comment]
@bmhm Can you rebase to resolve conflicts please?

**Comment by bmhm** [comment]
> @bmhm Can you rebase to resolve conflicts please?

Done, please re-review.

**Pull request merged**